src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/CaseNoteFilter.java

```diff
@@ -59,9 +78,39 @@ public Predicate toPredicate(@NotNull final Root<OffenderCaseNote> root, @NotNul
             predicateBuilder.add(cb.lessThan(root.get("occurrenceDate"), endDate.plusDays(1)));
         }
 
+        if(typesSubTypes != null){
+           predicateBuilder.add(getTypesPredicate(root, cb));
+        }
+
         final var predicates = predicateBuilder.build();
         return cb.and(predicates.toArray(new Predicate[0]));
+    }
+
+    private Predicate getTypesPredicate(final Root<OffenderCaseNote> root, final CriteriaBuilder cb) {
+       final var typesAndSubTypes = QueryParamHelper.splitTypes(typesSubTypes);
+       final var typesPredicates= typesAndSubTypes.entrySet()
+            .stream()
+            .map(typeSet -> {
+                if(typeSet.getValue().isEmpty()){
+                    return cb.equal(root.get("type").get("code"), typeSet.getKey());
+                }
+                else{
+                    return getSubtypesPredicate(root, cb, typeSet.getKey() ,typeSet.getValue());
+                }
+            })
+            .toList();
+        return cb.or(typesPredicates.toArray(new Predicate[0]));
```

Comment:
```suggestion
            });
        return cb.or(typesPredicates.toArray(Predicate[]::new));
```
don't need to convert into a list, can leave as a `Stream` and do a `toArray` on that